supplementary/libalica-turtlesim/src/WaitForMsg.cpp

```diff
@@ -11,25 +13,24 @@ WaitForMsg::WaitForMsg(alica::BehaviourContext& context)
 void WaitForMsg::initialiseParameters()
 {
     alica::LockedBlackboardRO bb{*getBlackboard()};
-    _topic = bb.get<std::string>("topic");
-    ROS_INFO_STREAM_NAMED(__func__, "Waiting for msg on " << _topic);
-    _sub = ros::NodeHandle("~").subscribe(_topic, 1, &WaitForMsg::onMsg, this);
+    auto turtle = alica::LockedBlackboardRO(*getGlobalBlackboard()).get<std::shared_ptr<turtlesim::TurtleInterfaces>>("turtle");
+    turtle->subOnMsg(bb.get<std::string>("topic"), getGlobalBlackboard());
```

Comment:
I think its better to have seperate ros1 & ros2 implementations of waitForMsg & waitForTrigger behaviours, that way the behaviour will also be same as before. Now the subscriber lives on even after the behaviour is destructed & the global blackboard is used instead of the local one